ethsigner/signer/multiplatform/src/main/java/tech/pegasys/ethsigner/signer/multiplatform/SigningMetadataTomlConfigLoader.java

```diff
@@ -0,0 +1,105 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.toml.TomlParseResult;
+
+class SigningMetadataTomlConfigLoader {
+
+  public static final String FILE_BASED_SIGNER = "file-based-signer";
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final String CONFIG_FILE_EXTENSION = ".toml";
+  private static final String GLOB_CONFIG_MATCHER = "**" + CONFIG_FILE_EXTENSION;
+
+  private final Path tomlConfigsDirectory;
+
+  SigningMetadataTomlConfigLoader(final Path rootDirectory) {
+    this.tomlConfigsDirectory = rootDirectory;
+  }
+
+  Optional<FileBasedSigningMetadataFile> loadMetadataForAddress(final String address) {
+    final List<FileBasedSigningMetadataFile> matchingMetadata =
+        loadAvailableSigningMetadataTomlConfigs().stream()
+            .filter(toml -> toml.getFilename().toLowerCase().endsWith(normalizeAddress(address)))
+            .collect(Collectors.toList());
+
+    if (matchingMetadata.size() > 1) {
+      LOG.error("Found multiple signing metadata TOML file matches for address {}", address);
+      return Optional.empty();
+    } else if (matchingMetadata.isEmpty()) {
+      return Optional.empty();
+    } else {
+      return Optional.of(matchingMetadata.get(0));
+    }
+  }
+
+  Collection<FileBasedSigningMetadataFile> loadAvailableSigningMetadataTomlConfigs() {
+    final Collection<FileBasedSigningMetadataFile> metadataConfigs = new HashSet<>();
+
+    try (final DirectoryStream<Path> directoryStream =
+        Files.newDirectoryStream(tomlConfigsDirectory, GLOB_CONFIG_MATCHER)) {
+      for (final Path file : directoryStream) {
+        getMetadataInfo(file).ifPresent(metadataFile -> metadataConfigs.add(metadataFile));
+      }
+      return metadataConfigs;
+    } catch (final IOException e) {
+      LOG.warn("Error searching for signing metadata TOML files", e);
+      return Collections.emptySet();
+    }
+  }
+
+  Optional<FileBasedSigningMetadataFile> getMetadataInfo(final Path file) {
+    try {
+      final TomlParseResult result =
+          TomlConfigFileParser.loadConfigurationFromFile(file.toAbsolutePath().toString());
+      final String type = result.getTable("signing").getString("type");
+      if (FILE_BASED_SIGNER.equals(type)) {
+        final String keyFilename = result.getTable("signing").getString("key-file");
+        final Path keyPath = new File(keyFilename).toPath();
+        final String passwordFilename = result.getTable("signing").getString("password-file");
+        final Path passwordPath = new File(passwordFilename).toPath();
+        return Optional.of(
+            new FileBasedSigningMetadataFile(file.getFileName().toString(), keyPath, passwordPath));
```

Comment:
nit: I personally would pull this out to a function "createFileSignerFrom(toml)" - its overkill now - but makes it super clean for the next phase.

Reply:
done